### PR TITLE
Config adjustments

### PR DESCRIPTION
**Pull request by schoenkaft**
This changes `SIMPLY_STATIC_DEPLOY_AWS_CREDENTIALS` to `SIMPLY_STATIC_DEPLOY_CONFIG`, because:

- The `url` entry isn't an AWS credential, it's a config thingy used in the plugin so the name is already a bit misleading.
- I think I'd still prefer to have one config/settings constant instead of lots of different ones (and constants are the way you do this in WordPress), so moving the `url` one to a single constant is not the path I'd like to take.
- I see keys added to the config in the (near) future, in that case it would already be a breaking change. If the constant is the same we could more easily show features based on existence of keys/properties without breaking functionality.

Does this make sense? If so:

- I moved the AWS parts to a `aws` entry. This could also allow us to add another provider in the future, like `digitalocean` and then point to it via `'provider' => 'digitalocean'`. Although there might be more proper solutions of course.
- I moved the few keys that were made optional to a `OPTIONAL_FIELDS` const. It's not used/needed, but felt okay to have as a reference. But we can definitely ditch it or change it. 
- I'm not passing `$config->aws` into the AWS-related constructors, since it expect a `Config` object. Although this works, it might also make sense to create/pass something like a `AwsConfig` object?
- This change now breaks the `Config` validator and magic getter. But before spending a lot of time there, maybe discuss this whole thing?

PS: added a few discardable syntax tweaks, to get all the classes a bit more in line.

### CODE REVIEW

**Comment by harmenjanssen** [comment]
This makes total sense. 👍 
All the changes in this PR look good to me. Indeed, we'll have to do some work on the `Config` object to make it work with nested structures.

I can imagine the `__get` returning the array wrapped in a new `Config` object like so:

```php
public function __get(string $key): string {
    $out = $this->data[$key] ?? '';
    return is_array($out) 
      ? new Config($out)
      : $out;
}
```

To provide a fluent interface.

The validation is a little trickier off the top of my head, but maybe we can pass in the required fields through the constructor. In that case validate can recursively call itself on the nested structure with the nested required field config.
It also solves the same problem in the above `__get`, where the new `Config` object wouldn't know how to validate its contents.

Do you want me to take a stab at it?

**Comment by schoenkaft** [comment]
> I can imagine the `__get` returning the array wrapped in a new `Config` object like so:
> 
> ```
> public function __get(string $key): string {
>     $out = $this->data[$key] ?? '';
>     return is_array($out) 
>       ? new Config($out)
>       : $out;
> }
> ```
> 
> To provide a fluent interface.

Yeah, sound neat! I'd prefer to inject scoped Configs into classes, and not have to fetch props from a global config inside those scoped classes, this would definitely solve that.

> The validation is a little trickier off the top of my head, but maybe we can pass in the required fields through the constructor. In that case validate can recursively call itself on the nested structure with the nested required field config.
> It also solves the same problem in the above `__get`, where the new `Config` object wouldn't know how to validate its contents.

Make sense, however a few points:

- I'm not sure if the more 'ad-hoc' en distributed 'required fields' approach will turn out in practice. It will definitely work, but then there will be no 'one place' where all required fields are found. Plus you'll get those errors only when constructing/invoking the method, so the validation errors are more tied to actions from the user (like a REST call triggering the build).
- This thing now throws an `Exception`, resulting in white screens. We also have the requirements check built by @HammenWS, which 'shows' more user friendly errors, and already checks for the presence of the constant itself. Feels we now have two separate ways to validate if 'requirements' are met. Although proper failing at 'runtime' is of course a good thing as well.
- So it might also be an idea to validate the configuration there, although I think the Interface/setup might not be suitable for that yet. But I would see something like this in the `SimplyStaticDeploy` class working as well:

```php
...->validate($config, ['url', 'aws']);
...->validate($config->aws, ['key', 'secret', 'region', 'bucket']);
```

Just a thought though...

> Do you want me to take a stab at it?

If you have time and 'zin', please do!

**Comment by harmenjanssen** [comment]
That's a good point, I will open up the validation methods from the `Config` object so Ramiro's validation code can actually introspect it.
In that it doesn't have to throw an exception per se, but returning `null` or something might be fine.

> If you have time and 'zin', please do!

I'll take a look, and consider the above points!

**Comment by schoenkaft** [comment]
Waiting for #17, then I'd say this one is ready for a final check...?

File: includes/Grrr/SimplyStaticDeploy/SimplyStaticDeploy.php [hunk 1]
```diff
@@ -1,24 +1,39 @@
 <?php namespace Grrr\SimplyStaticDeploy;
 
+use Grrr\SimplyStaticDeploy\Config\RequiredFields;
```
Comment (HammenWS) [inline includes/Grrr/SimplyStaticDeploy/SimplyStaticDeploy.php]: Deze `use` is hier niet nodig.

**Pull request merged**